docs/proposals/federated-ruler.md

```diff
@@ -0,0 +1,82 @@
+More explicitly, for a given recording rule and the produced series `foobarbaz` which is owned by the composite tenant `0|1|2|3`, a subtenant `0`, `1`, `2` or `3` is chosen to always store the series `foobarbaz` in, lets say `2`. Another rule and produced series `fizzbuzz` owned by the same composite tenant `0|1|2|3` makes this same chose of subtenant, in our case chosing `0`.
+
+## Challenges
+
+### Allow federated rules behind feature flag
+
+#### Challenge
+
+Federated tenant rules and alerts will not be a good fit for organization and should be behind a feature flag.
+
+#### Proposal
+
+For federated rules owned by a single tenant, creation of federated rules (those sourcing data from multiple tenants) should be blocked behind the feature flag `ruler.enable_federated_rules`
+
+To support composite tenants, if tenant federation is enabled for ruler and alertmanager via `tenant-federation.enabled`, then ruler use a `mergeQueryable` to aggregate the results of querying multiple tenants. The ruler and alertmanager APIs should be updated to always call `tenant.GetTenantIDs` instead of `tenant.GetTenantID`, which will use the MultiTenantResolver when tenant federation is enabled.
+
+### Allow federated rules only for select tenants
+
+#### Challenge
+
+For many organizations, the ability for any tenant to write a rule querying any other tenant is not acceptable and more fine grained control is required
+
+#### Proposal
+
+Since the current default is that a tenant should only be able to write rules against itself, we suggest a config option `ruler.allowed-federated-tenants`, a string slice of OrgIDs like `infra` or `0|1|2|3|4` which are allowed to write rules against all tenants. If a tenant `bar` attempts to create a federated rule, an error should be returned by the ruler api. Similarly an option `ruler.disallowed-federated-tenants` explicitly states a list of tenants for which federated rules are not allowed. Combining these in a `util.AllowedTenants` should allow us to quickly determine if federation is enabled or disabled for a given tenant at rule creation.
+
+### Where to store resulting series of federated rule
+
+#### Challenge
+
+A single tenant rule always stores produced series in the tenant where the rule exists. This 1 -> 1 mapping becomes a many -> 1 mapping for federated rules.
+
+#### Proposal
+
+As some use cases will demand that a specific federated rule, querying tenants `B` and `C`, is stored in the owning team's tenant `A`, an option to allow explicit assignment of destination tenant for a federated rule is needed. Some use cases where a set of tenants `A|B|C|...|Z` are being treated as a single logical tenant when querying this explicit assignment of destination tenant is not needed.
```

Comment:
What if we simply store the resulting series in the tenant owing the ruler config? So if tenant A configures a rule to query tenants B+C, then the resulting series will be stored in tenant A. If you want that result to be stored in tenant B instead, you move the rule to the tenant B rulers config.

Reply:
Going with @pracucci suggestion here